mat64/dense.go

```diff
@@ -360,6 +356,12 @@ func (m *Dense) Clone(a Matrix) {
 				copy(mat.Data[i*c:(i+1)*c], amat.Data[i*amat.Stride:i*amat.Stride+c])
 			}
 		}
+	case *Vector:
+		amat := aU.mat
+		mat.Data = make([]float64, aU.n)
```

Comment:
One more question: the default case reuses the slice because there is no way (unlike with *Vector or RawMatrixer) that such matrix could shadow a Dense, right?